dissect/target/plugins/general/users.py

```diff
@@ -16,6 +18,10 @@ class UsersPlugin(InternalPlugin):
 
     __namespace__ = "user_details"
 
+    def __init__(self, target: Target):
+        super().__init__(target)
+        self.find = lru_cache(4096)(self.find)
```

Comment:
```suggestion
        self.find = lru_cache(64)(self.find)
```
Shouldn't matter too much, but maybe a more sensible amount? Or maybe even 32?